src/content/content.js

```diff
@@ -1,20 +1,51 @@
 /* global browser */
 import Message from '@/lib/messaging/message/message.js'
+import ContentReadyMessage from '@/lib/messaging/message/content-ready-message.js'
+import EmbedLibMessage from '@/lib/messaging/message/embed-lib-message.js'
 import StateMessage from '@/lib/messaging/message/state-message'
 import MessagingService from '@/lib/messaging/service.js'
-import { TabScript, UIController, ExtensionSyncStorage } from 'alpheios-components'
+import { TabScript, UIController, ExtensionSyncStorage, HTMLPage } from 'alpheios-components'
 import ComponentStyles from '../../node_modules/alpheios-components/dist/style/style.min.css' // eslint-disable-line
 import StateResponse from '../lib/messaging/response/state-response'
 
 let messagingService = null
 let uiController = null
 
+let sendContentReadyToBackground = function sendContentReadToBackground () {
+  messagingService.sendMessageToBg(new ContentReadyMessage(uiController.state))
+    .catch((error) => console.error('Unable to send content ready message to background', error))
+}
+
+let sendStateToBackground = function sendStateToBackground () {
+  messagingService.sendMessageToBg(new StateMessage(uiController.state))
+    .catch((error) => console.error('Unable to send state to background', error))
+}
+
+let sendResponseToBackground = function sendResponseToBackground (request, state = uiController.state) {
+  messagingService.sendResponseToBg(new StateResponse(request, state)).catch(
+    (error) => {
+      console.error('Unable to send a response to a state request', error)
+    }
+  )
+}
+
 /**
  * State request processing function.
  */
 let handleStateRequest = function handleStateRequest (request) {
   let requestState = TabScript.readObject(request.body)
   let diff = uiController.state.diff(requestState)
+  // Set script state to what controller wants it to be
+  // uiController.state.update(requestState)
```

Comment:
if we don't need this, remove it from comments.

Reply:
maybe this needs to be back in to keep the panel status the correct state after a page reload/navigation?

Reply:
We don't need it here because we'll update state selectively, depending on whether it is an activation or deactivation request or whether it is some other change. Working will for me without it, so I will remove it.